src/main/java/Duke.java

```diff
@@ -1,10 +1,140 @@
 public class Duke {
+    static int taskCount = 0;
+    static Task[] taskList = new Task[100];
+    static final String COMMANDS = "Commands:\n    todo taskName\n    deadline deadlineName /by time\n" 
+            + "    event eventName /at time\n    list\n    done taskNumber\n    help\n    bye\n";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        displayWelcomeMessage();
+        inputAndExecuteCommand();
+        displayExitMessage();
+    }
+
+    private static void displayExitMessage() {
+        String exitMessage = "Sad to see you go! ): See you soon!";
+        printWithBorder(exitMessage);
+    }
+
+    private static void inputAndExecuteCommand() {
+        String line;
+        Scanner scanner = new Scanner(System.in);
+        
+        while (true) {
+            line = scanner.nextLine();
+            String[] commandTypeAndArg = line.split(" ", 2);
+            String commandType = commandTypeAndArg[0];
+            String commandArg = null;
+            if (commandTypeAndArg.length > 1) {
+                commandArg = commandTypeAndArg[1];
+            }
+
+            if (commandType.equals("bye")) {
+                scanner.close();
+                return;
+            }
+            executeCommand(commandType, commandArg);
+        }
+    }
+    
+    private static void executeCommand(String commandType, String commandArg) {
+        switch (commandType) {
+        case "help":
+            printWithBorder(COMMANDS);
+            break;
+        case "list":
+            listAllTasks();
+            break;
+        case "done":
+            markTaskAsDone(commandArg);
+            break;
+        case "todo":
+            addTodo(commandArg);
+            break;
+        case "deadline":
+            addDeadline(commandArg);
+            break;
+        case "event":
+            addEvent(commandArg);
+            break;
+        default:
+            displayInvalidCommandResponse();
+            break;
+        }
+    }
+
+    private static void addTodo(String commandArg) {
+        Todo task = new Todo(commandArg);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addEvent(String commandArg) {
+        String[] taskDescriptionAndAt = commandArg.split(" /at ", 2);
+        String description = taskDescriptionAndAt[0];
+        String at = taskDescriptionAndAt[1];
+        Event task = new Event(description, at);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addDeadline(String commandArg) {
+        String[] taskDescriptionAndBy = commandArg.split(" /by ", 2);
+        String description = taskDescriptionAndBy[0];
+        String by = taskDescriptionAndBy[1];
```

Comment:
description and by variables appropriately named

Reply:
thank you!